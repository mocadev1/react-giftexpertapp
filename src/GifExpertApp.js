import React, { useState } from 'react'
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = () => {

    // const categories = ['One Punch', 'Shippuden', 'Darling in the Franxx'];
    const [categories, setCategories] = useState(['Darling in the Franxx']);

    // const handleAdd = ( ) => setCategories( [...categories, 'category'] ); // Mi soluciÃ³n a la tarea
    /** 
     * El setCategories que retorna el useState puede recibir un callback como argumento,
     * a este callback se le pasa como argumento el estado actual del componente
     */
    // const handleAdd = ( ) => setCategories( cats => [...cats, 'category'] );
    
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr/>

            <ol>
                {
                    categories.map( category => (
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    ))
                }
            </ol>
        </>
    );
}

export default GifExpertApp
