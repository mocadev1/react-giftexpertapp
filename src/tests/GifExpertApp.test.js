import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';


describe('Pruebas en <GifExpertApp />', () => {
   
    test('debe de renderizarse correctamente', () => {
       
        const wrapper = shallow( <GifExpertApp /> );

        expect( wrapper ).toMatchSnapshot();
        
        
        
    });
    

    test('debe de mostrar una lista de categorias', () => {
       
        const categories = ['Shippuden', 'Demon Slayer', 'Promised Neverland'];

        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );
        
        
        
    });
    
    
    
    
});
