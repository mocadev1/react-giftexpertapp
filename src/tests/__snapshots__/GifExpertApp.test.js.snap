// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Shippuden"
      key="Shippuden"
    />
    <GifGrid
      category="Demon Slayer"
      key="Demon Slayer"
    />
    <GifGrid
      category="Promised Neverland"
      key="Promised Neverland"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debe de renderizarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
